library/Zend/Http/Client.php

```diff
@@ -1592,6 +1592,11 @@ protected function _validateHeaderValue($value, $recurse = true)
             return;
         }
 
+        // Cast integers and floats to strings for purposes of header representation.
+        if (is_int($value) || is_float($value)) {
+            $value = (string) $value;
+        }
+
         if (! is_string($value) && (! is_object($value) || ! method_exists($value, '__toString'))) {
```

Comment:
As noted in IRC, the only other scalar types we can safely cast to string are floats and integers. Booleans should not be cast.